modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-evaluator-impl/src/main/java/com/liferay/dynamic/data/mapping/form/evaluator/internal/helper/DDMFormEvaluatorHelper.java

```diff
@@ -399,27 +438,42 @@ protected DDMFormFieldValidation getDDMFormFieldValidation(
 		return disabledPagesIndexes;
 	}
 
-	protected boolean isFieldEmpty(
-		DDMFormEvaluatorFieldContextKey ddmFormFieldContextKey) {
+	protected boolean isConfirmationValueInvalid(
+		DDMFormEvaluatorFieldContextKey ddmFormEvaluatorFieldContextKey) {
 
-		DDMFormFieldValue ddmFormFieldValue =
-			_ddmFormEvaluatorFormValuesHelper.getDDMFormFieldValue(
-				ddmFormFieldContextKey);
+		DDMFormFieldValue ddmFormFieldValue = getDDMFormFieldValue(
+			ddmFormEvaluatorFieldContextKey);
 
 		if (ddmFormFieldValue == null) {
-			return true;
+			return false;
 		}
 
-		Object value =
-			ddmFormEvaluatorDDMExpressionFieldAccessor.getFieldPropertyChanged(
-				ddmFormFieldContextKey, "value");
+		String confirmationValue = Objects.toString(
+			ddmFormFieldValue.getConfirmationValue(), StringPool.BLANK);
 
-		if (value != null) {
-			updateDDMFormFieldValue(ddmFormFieldValue, value);
+		Value value = ddmFormFieldValue.getValue();
+
+		if (value == null) {
+			return !Objects.equals(confirmationValue, StringPool.BLANK);
+		}
+
+		return !Objects.equals(
+			confirmationValue,
+			value.getString(_ddmFormEvaluatorEvaluateRequest.getLocale()));
+	}
```

Comment:
Hi @natocesarrego, It would be better to do something like:

```
if(Objects.equals(confirmationValue, value)) {
	return false;
}

return true;
```
?